### PR TITLE
use varuint-bitcoin

### PR DESCRIPTION
**Pull request by dcousens**
Tests are failing due to different error messages.
Not sure if its a breaking change or not,  tagging as such for now until we discuss otherwise.

Closes https://github.com/bitcoinjs/bitcoinjs-lib/issues/565

### CODE REVIEW

File: src/bufferutils.js [hunk 1]
```diff
@@ -174,8 +119,8 @@ module.exports = {
   readUInt64LE: readUInt64LE,
   readVarInt: readVarInt,
   reverse: require('buffer-reverse'),
-  varIntBuffer: varIntBuffer,
-  varIntSize: varIntSize,
+  varIntBuffer: varuint.encode,
+  varIntSize: varuint.encodingLength,
```
Comment (dcousens) [inline src/bufferutils.js]: @fanatid its worth asking if this should be included.
I'd say these functions are non-generic enough that it makes sense to keep them in this library.

**Comment by dcousens** [comment]
ping @fanatid

**Comment by jprichardson** [comment]
> ping @fanatid 

FYI. I think he's gone until late July.

File: src/bufferutils.js (hunk 1 above)
Comment (fanatid) [inline src/bufferutils.js]: I think they should be included for backward compatibility (at least while 3.x not be released).

Comment (dcousens) [inline src/bufferutils.js]: This PR is breaking anyway, I meant including them in `3.x.`

**Comment by dcousens** [comment]
Ready to merge

**Comment by dcousens** [comment]
Not a breaking change, I've kept the functions internally

**Comment by dcousens** [comment]
https://travis-ci.org/bitcoinjs/bitcoinjs-lib/jobs/165689761 was the only failed job,  with the reason:

>  1) bitcoinjs-lib (advanced) can create an OP_RETURN transaction:
     Error: timeout of 20000ms exceeded. Ensure the done() callback is being called in this test.

I'll consider extending the timeouts I suppose.

**Pull request merged**